src/Messages/OWSMessageSender.m

```diff
@@ -42,10 +42,12 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-#define SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(major, minor)                                                          \
-    ([[NSProcessInfo processInfo]                                                                                      \
-        isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){                                                    \
-                                            .majorVersion = major, .minorVersion = minor, .patchVersion = 0 }])
+void AssertIsOnSendingQueue()
+{
+    if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(10, 0)) {
+        dispatch_assert_queue([OWSDispatch sendingQueue]);
```

Comment:
It's not always safe to assert the current queue. Sometimes when you dispatch_sync onto another queue you don't actually switch queues as an optimization. 

From https://developer.apple.com/reference/dispatch/1452870-dispatch_sync

> As an optimization, this function invokes the block on the current thread when possible.

So this is only safe if we're using dispatch_async(), etc.  And I tend to only assert that we're on the main queue since a) I strenuously avoid doing dispatch_sync() onto the main queue. b) even if this is safe today, tomorrow someone might dispatch_sync() to code with this assertion.